### PR TITLE
Add TrySingle overloads for 

### PR DESCRIPTION
**Pull request by elianora**
Addresses #631 by adding and unit testing the following:
```csharp
public static TSource? TrySingle<TSource>(this IEnumerable<TSource> source);
public static ValueTask<TSource?> TrySingle<TSource>(this IAsyncEnumerable<TSource> source, CancellationToken cancellationToken = default)
```

### CODE REVIEW

File: Source/SuperLinq/TrySingle.cs [hunk 1]
```diff
@@ -112,4 +112,43 @@ public static (TCardinality Cardinality, T? Value)
...
+	{
+		ArgumentNullException.ThrowIfNull(source);
+
+		if (source.TryGetCollectionCount() is int n)
+		{
+			return n switch
+			{
+				0 => default,
+				1 => source.First(),
+				_ => default
```
Comment (viceroypenguin) [inline Source/SuperLinq/TrySingle.cs:139]: can be simplified:
```suggestion
				1 => source.First(),
				0 or _ => default
```

**Review by viceroypenguin — COMMENTED** [review]
One minor nit. Looks great though! Thank you!

File: Source/SuperLinq/TrySingle.cs (hunk 1 above)
Comment (elianora) [reply Source/SuperLinq/TrySingle.cs]: Done!

**Comment by viceroypenguin** [comment]
Build errors aren't your fault. I'm gonna fix those real quick and then I'll rebuild your PR.

File: Tests/SuperLinq.Test/TrySingleTest.cs [hunk 2]
```diff
@@ -110,4 +110,15 @@ public void TrySingleEnumeratesOnceOnlyAndDisposes(int numberOfElements, string
 		var (cardinality, _) = seq.TrySingle("zero", "one", "many");
 		Assert.Equal(expectedCardinality, cardinality);
 	}
+
+	[Theory]
+	[InlineData(0, 0)]
+	[InlineData(1, 1)]
+	[InlineData(2, 0)]
```
Comment (viceroypenguin) [inline Tests/SuperLinq.Test/TrySingleTest.cs:117]: Code coverage pointed out that we're not testing collection branch. Please make a copy of `GetSequences` that returns a list of sequences and expected results. This way we can test both branches, for 0, 1, many.

**Pull request merged**